src/mmfatl.c

```diff
@@ -245,55 +239,82 @@ static void initState(struct ParserState* state, struct Buffer* buffer) {
   state->format = empty;
 }
 
-/*! reflect a possible buffer overflow in the parser state
+/*!
+ * \brief convert an unsigned to a string of decimal numbers
+ *
+ * Converts an unsigned to a sequence of decimal digits representing its value.
+ * The value range is known to be at least 2**32 on contemporary hardware, but
+ * C99 guarantees just 2**16.  We support unsigned in formatted error output
+ * to allow for macros like __LINE__ denoting error positions in text files.
+ *
+ * There exist no utoa in the C99 standard library, that could be used instead,
+ * and sprintf must not be used in a memory-tight situation (AS Unsafe heap,
+ * https://www.gnu.org/software/libc/manual/html_node/Formatted-Output-Functions.html).
+ * \param value an unsigned value to convert.
+ * \returns a pointer to a string converted from \p value.  Except for zero,
+ *   the result has a leading non-zero digit.
+ * \attention  The result is stable only until the next call to this function.
```

Comment:
This should be clearer about the ownership of the data. I assume from this line that the return value references a buffer owned by this function, which should not be deallocated by the caller.

Reply:
You're right.  Each call to this function overwrites the previous result held in a local buffer